### DISCUSSION TITLE
<title>Transferring repositories from GitHub to Github

### DISCUSSION THREAD

**Discussion by YLohithReddy** [community/community#55916] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Transferring repositories from GitHub to UCSF Github, with full commit and branch history intact?

**Comment by nirav-gajera** [comment, marked as answer]
1.Create a new repository on UCSF GitHub: Start by creating an empty repository on UCSF GitHub. Note down the repository's URL
2. Clone source repo- git clone --mirror <source_repository_url>
3. Change remote URL - cd <cloned_repository_directory>
      git remote set-url --push origin <ucsf_github_repository_url>
4.Push repo to UCSF Github - git push --mirror
5. Verify the transfer: Once the push is complete, go to the UCSF GitHub repository's page

**Reply by YLohithReddy** [reply]
Thank you @nirav-gajera
